src/main/java/edu/ucla/library/bucketeer/verticles/SlackMessageWorkerVerticle.java

```diff
@@ -0,0 +1,154 @@
+
+package edu.ucla.library.bucketeer.verticles;
+
+import static edu.ucla.library.bucketeer.Constants.MESSAGES;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.io.FileUtils;
+
+import com.github.seratch.jslack.Slack;
+import com.github.seratch.jslack.api.methods.SlackApiException;
+import com.github.seratch.jslack.api.methods.response.files.FilesUploadResponse;
+import com.github.seratch.jslack.api.webhook.Payload;
+import com.github.seratch.jslack.api.webhook.WebhookResponse;
+
+import info.freelibrary.util.Logger;
+import info.freelibrary.util.LoggerFactory;
+
+import edu.ucla.library.bucketeer.Constants;
+import edu.ucla.library.bucketeer.HTTP;
+import edu.ucla.library.bucketeer.MessageCodes;
+import edu.ucla.library.bucketeer.Op;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+/**
+ * Accepts and forwards messages to a configured Slack channel.
+ */
+public class SlackMessageWorkerVerticle extends AbstractBucketeerVerticle {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SlackMessageWorkerVerticle.class, MESSAGES);
+
+    @Override
+    public void start() {
+        final JsonObject config = config();
+
+        if (LOGGER.isDebugEnabled()) {
+            final String threadName = Thread.currentThread().getName();
+            final String className = SlackMessageWorkerVerticle.class.getSimpleName();
+
+            LOGGER.debug(MessageCodes.BUCKETEER_004, className, threadName);
+        }
+
+        // set up the Slack client
+        LOGGER.debug(MessageCodes.BUCKETEER_085);
+
+        final Slack slack = Slack.getInstance();
```

Comment:
Just noting here that I think JSlack is a blocking library. This should be fine since we're sending small amounts of data, but it may be something that's changeable (to a non-blocking solution) in the future. If we see weirdness or performance issues we could also turn this verticle into a worker verticle, fwiw.

Reply:
agreed, save this for a future refactor